cpp/src/arrow/compute/exec/source_node.cc

```diff
@@ -308,9 +308,6 @@ struct SchemaSourceNode : public SourceNode {
     auto& schema = cast_options.schema;
     auto io_executor = cast_options.io_executor;
 
-    if (io_executor == NULLPTR) {
-      io_executor = plan->exec_context()->executor();
-    }
```

Comment:
Is this removal intended?

Reply:
It's not terribly relevant.  I was initially wanting to use `exec_batch_source` here but ran into a problem because I was not transferring off the background generator.  Incidentally, I think it may be close to time to fix the background generator to remove this limitation, but I have enough on my plate for the moment.

This change was simply because, a few lines down (on line 316) we have a very similar if condition:

```
    if (io_executor == NULLPTR) {
      io_executor = io::internal::GetIOThreadPool();
    }
```

I don't think we need both of these statements and defaulting to the I/O thread pool seemed like the better default.  If that is not the correct default, or there is some subtlety I am missing, let me know and I can revert this.